internal/provider/resource_deployment.go

```diff
@@ -26,233 +25,244 @@ func resourceDeployment() *schema.Resource {
+				Description: "Node id to place the deployment on.",
 			},
 			"name": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Default:  "vm",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "vm",
+				Description: "Solution name for created contract, displayed [here](https://play.dev.grid.tf/#/contractslist).",
 			},
 			"solution_type": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Default:  "Virtual Machine",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "Virtual Machine",
+				Description: "Solution type for created contract, displayed [here](https://play.dev.grid.tf/#/contractslist).",
 			},
 			"solution_provider": {
 				Type:        schema.TypeInt,
 				Optional:    true,
 				Default:     0,
-				Description: "Solution provider ID",
+				Description: "Solution provider ID for the deployed solution which allows the creator of the solution to gain a percentage of the rewards.",
 			},
 			"ip_range": {
 				Type:        schema.TypeString,
 				Computed:    true,
-				Description: "IP range of the node (e.g. 10.1.2.0/24)",
+				Description: "IP range of the node for the wireguard network (e.g. 10.1.2.0/24). Has to have a subnet mask of 24.",
 			},
 			"network_name": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				Description: "Network to use for Zmachines",
+				Description: "Network name of the deployed network resource to connect vms.",
 			},
 			"disks": {
-				Type:     schema.TypeList,
-				Optional: true,
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of disk workloads configurations.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"name": {
 							Type:        schema.TypeString,
 							Required:    true,
-							Description: "the disk name, used to reference it in zmachine mounts",
+							Description: "Disk workload name.",
 						},
 						"size": {
 							Type:        schema.TypeInt,
 							Required:    true,
-							Description: "the disk size in GBs",
+							Description: "Disk size in GBs.",
 						},
 						"description": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "",
+							Type:        schema.TypeString,
+							Optional:    true,
+							Default:     "",
+							Description: "Description for disk workload.",
 						},
 					},
 				},
 			},
 			"zdbs": {
-				Type:     schema.TypeList,
-				Optional: true,
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of Zdb workloads configurations. You can read more about 0-db (Zdb) [here](https://github.com/threefoldtech/0-db/).",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"name": {
-							Type:     schema.TypeString,
-							Required: true,
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Zdb worklod name.",
```

Comment:
for every name let's mention that the name needs to be unique within the deployment